activerecord/lib/active_record/associations/preloader.rb

```diff
@@ -117,12 +117,15 @@ def preloaders_for_hash(association, records, scope, polymorphic_parent)
 
             recs = loaders.flat_map(&:preloaded_records).uniq
 
-            reflection = records.first.class._reflect_on_association(parent)
-            polymorphic_parent = reflection && reflection.options[:polymorphic]
+            if first_record = records.first
+              reflection = first_record.class._reflect_on_association(parent)
+              polymorphic_parent = reflection && reflection.options[:polymorphic]
+
+              loaders.concat Array.wrap(child).flat_map { |assoc|
```

Comment:
I think this should be outside the if statement? at least it wasn't part of the change I added

Reply:
@robotdana the block passed to `flat_map` references `polymorphic_parent`